BDArmory/FX/ShellCasing.cs

```diff
@@ -20,12 +21,10 @@ public class ShellCasing : MonoBehaviour
         void OnEnable()
         {
             startTime = Time.time;
-            velocity = initialV;
-            velocity += transform.rotation *
-                        (new Vector3(Random.Range(-.1f, .1f), Random.Range(-.1f, .1f),  Random.Range(-.1f, .1f)) + configV);
-            angularVelocity =
-                new Vector3(Random.Range(-10f, 10f), Random.Range(-10f, 10f),
-                    Random.Range(-10f, 10f)) * 10;
+            Vector3 randV = Random.insideUnitSphere * configD;
+            velocity = initialV + transform.rotation *
+                        (new Vector3(configV.x * randV.x, configV.y  * randV.y, configV.z * randV.z) + configV);
```

Comment:
This formulation zeroes out the deviation in the axes where `configV` is 0.
Compare
![Screenshot_20240629_111946](https://github.com/BrettRyland/BDArmory/assets/18440126/6a35a8c9-a6d5-4533-9cd2-8af465620433)
![Screenshot_20240629_112012](https://github.com/BrettRyland/BDArmory/assets/18440126/90c9882f-e59e-483c-a0a6-bed37f205427)
vs
![Screenshot_20240629_112314](https://github.com/BrettRyland/BDArmory/assets/18440126/824f028e-5352-4445-81c0-1bfae4b5b710)
![Screenshot_20240629_112340](https://github.com/BrettRyland/BDArmory/assets/18440126/062b4089-7542-4eed-9d06-6caf4cb67c80)
where the former two are using your PR and the latter two are from modifying the PR to
```diff
diff --git a/BDArmory/FX/ShellCasing.cs b/BDArmory/FX/ShellCasing.cs
index 0bd154c2b..9a00d5dc7 100644
--- a/BDArmory/FX/ShellCasing.cs
+++ b/BDArmory/FX/ShellCasing.cs
@@ -11,0 +12,2 @@ public class ShellCasing : MonoBehaviour
+        public Vector3 configV;
+        public Vector3 configD;
@@ -17 +19 @@ public class ShellCasing : MonoBehaviour
-        const int collisionLayerMask = (int)(LayerMasks.Parts | LayerMasks.Scenery | LayerMasks.Unknown19 | LayerMasks.Wheels); // Why 19?
+        const int collisionLayerMask = (int)(LayerMasks.Parts | LayerMasks.Scenery | LayerMasks.EVA | LayerMasks.Wheels);
@@ -22,7 +24,7 @@ void OnEnable()
-            velocity = initialV;
-            velocity += transform.rotation *
-                        new Vector3(Random.Range(-.1f, .1f), Random.Range(-.1f, .1f),
-                            Random.Range(6f, 8f));
-            angularVelocity =
-                new Vector3(Random.Range(-10f, 10f), Random.Range(-10f, 10f),
-                    Random.Range(-10f, 10f)) * 10;
+            Vector3 randV = Random.insideUnitSphere;
+            velocity = initialV + transform.rotation * new Vector3(
+                configV.x + configD.x * randV.x,
+                configV.y + configD.y * randV.y,
+                configV.z + configD.z * randV.z
+            );
+            angularVelocity = 100f * Random.insideUnitSphere;
diff --git a/BDArmory/Weapons/ModuleWeapon.cs b/BDArmory/Weapons/ModuleWeapon.cs
index c755a1a4d..ae7b00eb8 100644
--- a/BDArmory/Weapons/ModuleWeapon.cs
+++ b/BDArmory/Weapons/ModuleWeapon.cs
@@ -399,0 +400,6 @@ public float GetEngageRange()
+        [KSPField]
+        public Vector3 shellEjectVelocity = new Vector3(0, 0, 7);
+
+        [KSPField]
+        public float shellEjectDeviation = 0.1f;
+
@@ -3324,0 +3331,6 @@ void WeaponFX()
+                        shellComponent.configV = shellEjectVelocity;
+                        shellComponent.configD = new Vector3(
+                            shellEjectDeviation * (1f + Mathf.Abs(shellEjectVelocity.x)),
+                            shellEjectDeviation * (1f + Mathf.Abs(shellEjectVelocity.y)),
+                            shellEjectDeviation * (1f + Mathf.Abs(shellEjectVelocity.z))
+                        );
```

Reply:
Since I haven't heard anything further, I'm going to merge this with the above implementation, which properly preserves the original behaviour while adding the new control of the principal ejection direction.